build.sh

```diff
@@ -10,16 +10,51 @@ SCRIPT_DIR=$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)
 ###########################################################################
 
 BUILD_PROJECT_FILE="$SCRIPT_DIR/build/_build.csproj"
-TEMP_DIRECTORY="$SCRIPT_DIR//.nuke/temp"
+TEMP_DIRECTORY="$SCRIPT_DIR/.nuke/temp"
 
-DOTNET_GLOBAL_FILE="$SCRIPT_DIR//global.json"
+DOTNET_GLOBAL_FILE="$SCRIPT_DIR/global.json"
 DOTNET_INSTALL_URL="https://dot.net/v1/dotnet-install.sh"
 DOTNET_CHANNEL="Current"
 
 export DOTNET_CLI_TELEMETRY_OPTOUT=1
 export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
 export DOTNET_MULTILEVEL_LOOKUP=0
 
+#### START CUSTOM CODE ####
+# This file is an odd one. `nuke :update` wants to make edits to it, so we want to keep it similar to the default 
+# as we can. But, we want to do some tricky stuff, and make sure both dotnet 6 and dotnet 7 are installed, so we
+# can compile for both versions. So... We've got our "custom code" block here, which we want to keep, and the 
+# default code down below so we can see what changes are made upstream, and apply them to ours if we need to.
+
+# Download install script
+DOTNET_INSTALL_FILE="$TEMP_DIRECTORY/dotnet-install.sh"
+mkdir -p "$TEMP_DIRECTORY"
+curl -Lsfo "$DOTNET_INSTALL_FILE" "$DOTNET_INSTALL_URL"
+chmod +x "$DOTNET_INSTALL_FILE"
+
+echo "Installing net6"
+"$DOTNET_INSTALL_FILE" --install-dir "$TEMP_DIRECTORY/dotnet" --channel "6.0" --no-path
+
+echo "Installing net7"
+"$DOTNET_INSTALL_FILE" --install-dir "$TEMP_DIRECTORY/dotnet" --channel "7.0" --no-path
```

Comment:
Doesn't Nuke bootstrapper already install based on global.json?

Reply:
It's supposed to, but it doesn't work so well - https://build.octopushq.com/buildConfiguration/OctopusDeploy_LIbraries_OctoVersion_Ci/7699202?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=7699202_946_899&logFilter=debug&logView=flowAware